src/Cosi-BLS/transaction.lisp

```diff
@@ -123,13 +123,16 @@ the simple commitment to the uncloaked value"
             :initarg :txins)
    (txouts  :reader  trans-txouts  ;; txout or integer, to new owner or self
             :initarg :txouts)
+   (fee     :reader  trans-fee
+            :initarg :fee
+            :initform 0)
    (gamadj  :reader  trans-gamadj  ;; A curve adjustment to get to zero balance.
             :initarg :gamadj) 
    ))
 
 ;; ---------------------------------------------------------------------
 
-(defun make-transaction (txins gam-txins txouts txout-secrets)
+(defun make-transaction (txins gam-txins txouts txout-secrets fee)
```

Comment:
@dbmcclain You need to change the method signature for `make-transaction` in `src/Cosi-BLS/t/cosi-tests.lisp`.  

Alternately, you could let current code continue to work via use of keyword arguments ala:

```
(defun make-transaction (txins gam-txis txouts txout-secrets &key (fee 0))
…
```

Reply:
thanks for catching that... I'm keeping fees as a required argument, instead of a default optional arg. Are there going to be any cases of fee-less transactions? I can change to your suggested form.

Reply:
Naw, go ahead with what you are comfortable. 

I tend to abhor `&optional` stuff in favor of `&key` with default values for two reasons:

1)  Upgrading code with `&optional` arguments to add/subtract arguments is a real messy job.

2) After a function gets more than a couple arguments, I find that having  `&key` parameters gives a much better mnemonic for users of the API.   And allows one to specify sensible defaults.

Reply:
yes, I can think of two cases for fee-less transactions. Those being when the fee is spent to participant witnesses or leader node, and when escrow amounts are forfeited. So we should probably change to an optional or, better, a keyword argument with default 0. I agree.